Sources/RxMoya/RxMoyaProvider.swift

```diff
@@ -21,7 +21,10 @@ open class RxMoyaProvider<Target>: MoyaProvider<Target> where Target: TargetType
 
         // Creates an observable that starts a request each time it's subscribed to.
         return Observable.create { [weak self] observer in
```

Comment:
I'm leaning towards having the provider stay weakly captured and producing an error if its been deallocated. I don't think it's the responsibility of the `request()` function to retain the provider, that falls on whoever is using it. And if the provider gets deallocated before the request completes, an error makes sense. As you said, users can work around this if they need to, so it provides the most flexibility.